set.mm

```diff
@@ -532760,6 +532528,73 @@ dependency is expressed in our hypothesis (called implicit
       CDBJABCDKL $.
   $}
 
+  ${
+    $d t x $.  $d x ph $.
+    $( Reverting substitution yields the original expression.  (Contributed by
+       Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich-lem1 $p |- ( [ t / x ] [ x / t ] ph <-> ph ) $=
+      ( wsb sbco2vv sbid bitri ) ACBDBCDACCDAACCBEACFG $.
+  $}
+
+  ${
+    $d a x $.  $d a b y $.  $d a b ph $.
+    $( Reverting two substitutions in succession.  (Contributed by Wolf Lammen,
+       5-Aug-2023.) $)
+    wl-ich-lem2 $p |- ( [ y / b ] [ x / a ] [ a / x ] [ b / y ] ph <-> ph ) $=
+      ( wsb wl-ich-lem1 sbbii bitri ) ACEFZBDFDBFZECFJECFAKJECJDBGHAECGI $.
+  $}
+
+  $( ~ spsbim applied twice.  (Contributed by Wolf Lammen, 5-Aug-2023.) $)
+  wl-ich-lem3 $p |- ( A. a A. b ( ph <-> ps ) ->
+       ( [ y / b ] [ x / a ] ph <-> [ y / b ] [ x / a ] ps ) ) $=
+    ( wb wal wsb alcom nfa1 sp sbbid sps sylbi ) ABGZFHEHPEHZFHZAECIZFDIBECIZFD
+    IGPEFJRSTFDQFKQSTGFQABECPEKPELMNMO $.
+
+  ${
+    $d a x $.  $d a b y $.  $d a b ph $.
+    $( Lemma.  (Contributed by Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich-lem4 $p |- ( A. a A. b ( [ a / x ] [ b / y ] ph
+                              <-> [ b / x ] [ a / y ] ph ) ->
+        ( ph <-> [ y / b ] [ x / a ] [ b / x ] [ a / y ] ph ) ) $=
+      ( wsb wb wal wl-ich-lem2 wl-ich-lem3 syl5bbr ) AACEFBDFZDBFECFLACDFBEFZGE
+      HDHMDBFECFABCDEILMBCDEJK $.
+  $}
+
+  ${
+    $d a b x $.  $d a b y $.  $d a b ph $.
+    $( Infer one criterium for interchangeability from another.  (Contributed
+       by Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich $p |- ( A. a A. b ( [ a / x ] [ b / y ] ph
```

Comment:
That's a great result, but I wonder if `a` in the conclusion can be changed to `z` (or any other variable not used in the antecedent), as also proposed by @benjub .

Reply:
Of course.  I did not read @benjub s comments before, but did an analysis on my own.

Reply:
Did not have the time to look at this detail.